runtime-sdk/modules/contracts/src/store.rs

```diff
@@ -13,12 +13,26 @@ use crate::{state, types, Error, MODULE_NAME};
 pub const CONFIDENTIAL_STORE_KEY_PAIR_ID_CONTEXT_BASE: &[u8] =
     b"oasis-runtime-sdk/contracts: state";
 
+const CONTEXT_CONFIDENTIAL_STORE_TX_INSTANCE_COUNT: &str = "oasis-runtime-sdk/Context store instance count";
+
 /// Create a contract instance store.
+///
+/// Confidential stores are only available in transaction contexts.
 pub fn for_instance<'a, C: Context>(
     ctx: &'a mut C,
     instance_info: &types::Instance,
     store_kind: StoreKind,
 ) -> Result<Box<dyn Store + 'a>, Error> {
+    let instance_count: usize = *ctx.value(CONTEXT_CONFIDENTIAL_STORE_TX_INSTANCE_COUNT).or_default();
```

Comment:
Should be ignored for non-confidential setups (e.g. move it down to where the confidential store is instantiated).